src/event/model/event.rs

```diff
@@ -46,7 +46,14 @@ pub enum Event {
 }
 
 impl Event {
-    pub async fn fetch(id: i32, pool: &MySqlPool) -> Result<Event, EventError> {
+    pub async fn fetch<'a, E>(id: i32, executor: E) -> Result<Event, EventError>
+    where
+        E: Executor<'a>,
+    {
+        let mut transaction = executor
+            .begin()
+            .await
+            .map_err(|inner| EventError::DatabaseError { inner })?;
```

Comment:
Queries use the same trait (so that we can use queries from mutations). We don't need to roll back (or commit) the transaction here since it didn't change anything in the database. Interesting side effect: this guarantees that `Event::fetch` doesn't modify the database at all.